src/Gateway/PayplugGateway.php

```diff
@@ -255,6 +255,10 @@ public function validate_payment()
 			return;
 		}
 
+		// Prevent the hook "the_post" from being called multiple times
+		if (did_action("the_post") >= 2)
```

Comment:
did_action always returns 1, you can change the_post action by "woocommerce_thankyou" action, ti makes more sens

Reply:
and also this can have some conflicts with other plugins that are using the same action hook.